prep'ing Dan's changes for a pull request to upstream

binary1230 opened the pull request:
We're preparing to try use this bot for Magfest to replace our current bot (based on Hubot, written in coffeescript) with this bot (written in python).

@migetman9 prototyped a bunch of new features to enhance the upstream branch, this pull request is going to serve to refine those features in order to get them in shape to submit an upstream pull request.  If not accepted upstream, we can at least continue to pull fixes from upstream into this fork.

@EliAndrewC, soon I would love your eyes on this for some legit code review as I'd like to start treating our bot code like we do the rest of our production code.

binary1230 committed:
updating with Dan Evan's changes, WIP

Commit:
closer to original

Commit:
stuff to make closer

Commit:
as close as I can to the original

binary1230 committed:
adding in extra Dan-specific files missing

binary1230 commented:
Our 'master' branch is upstream's 'develop' branch

binary1230 commented on .gitignore:
```diff
@@ -1,12 +1,12 @@
+
+.idea/snakeman.iml
+
+.idea/workspace.xml
```
@migetman9 we should restore the original stuff in this file, and just exclude the .idea directory instead of the individual things in it

binary1230 commented on README.md:
````diff
@@ -57,7 +57,7 @@ if __name__ == "__main__":
 ```python
-default_reply = "Sorry but I didn't understood you" 
+default_reply = "Sorry but I didn't understood you"
 ```
````
trivial thing: add the space back in this file so it disappears from the diff.  our goal is to have the minimal amount of changes to enhance readability in reviews like this one.

binary1230 commented on slackbot/VERSION:
```diff
@@ -1 +0,0 @@
-0.3.1
```
add this back in

binary1230 commented on slackbot/dispatcher.py:
```diff
@@ -100,20 +99,19 @@ def loop(self):
+        self._client.rtm_send_message(msg['channel'], default_reply)
 
-def unicode_compact(func):
+    def unicode_compact(func):
```
fix indent

binary1230 commented on slackbot/dispatcher.py:
```diff
@@ -205,6 +203,35 @@ def react(self, emojiname):
+                    tmpf.flush()
+                    tmpf.close()
+                self.channel.upload_file(title, f)
+
```
is upload_snippet() something generic enough that it should belong in the core here? (it might be)

daredoes replied:
upload_snippet is SO USEFUL

binary1230 commented on slackbot/settings.py:
```diff
@@ -1,33 +0,0 @@
-    try:
-        from local_settings import *
-    except ImportError:
-        pass
```
restore this file

daredoes replied:
Blegh, I get that it keeps us close to the original, but these files no longer belong here.

daredoes replied:
So I readded this file but it contains sensitive info, so I have it hidden.

binary1230 replied:
The file checked in here should not contain sensitive info, if it does we should remove the info and instead either use a custom run.py or pass the sensitive info via env vars

binary1230 commented on slackbot/utils.py:
```diff
@@ -10,6 +10,11 @@
+till_white = "([^\s]+)"
+till_end = "(.*)"
+issue = "(\\S[^\\-\\s]*-[*\\d])"
+j_key = "(\\S[^\\-\\s]*)"
```
are issue and j_key JIRA-related? if so, kill them here and we'll include this in the JIRA plugin

daredoes replied:
Sounds good

binary1230 commented on start.sh:
```diff
@@ -0,0 +1,5 @@
+cd plugins
+supervisorctl stop snakeman
+git pull
+supervisorctl start snakeman
```
probably kill this, we can do this part in another layer

daredoes replied:
can be killed, from the old setup

binary1230 commented on slackbot/bot.py:
```diff
@@ -7,22 +7,22 @@
             bot_emoji=settings.BOT_EMOJI if hasattr(settings, 'BOT_EMOJI') else None
         )
-        self._plugins = PluginsManager()
+        self._plugins = PluginsManager(settings)
```
see question below about restoring 'settings' here to the way it was before

binary1230 commented on slackbot/manager.py:
```diff
@@ -5,15 +5,14 @@
-    def __init__(self):
-        pass
+    def __init__(self, settings):
+        self.settings = settings
```
what's the rationale for passing in settings here vs the way it was being done before with `from slackbot import settings`?  (or is this an artifact of the earlier now-removed refactor where we had different bot settings)

daredoes replied:
To allow for multiple bots the Plugin Manager needed to be rewritten to read the plugins from the passed settings file, instead of grabbing a static one declared in this python script

binary1230 replied:
See above comment, this might not be needed if we are only running one bot out of this code. Multiple bots will run in different processes, each process can be started with different env vars

binary1230 commented on slackbot/manager.py:
```diff
@@ -55,9 +64,14 @@ def _load_plugins(self, plugin):
-            m = matcher.search(text)
-            if m:
+            m = matcher[0].search(text)
+            if m and has_answer_key:
```
what's the goal with the has_answer_key here?

binary1230 commented on slackbot/ears.py:
```diff
@@ -0,0 +1,25 @@
+        PluginsManager.commands['listen_to'][tuple((re.compile(matchstr, flags), halp))] = func
+        logger.info('registered listen_to plugin "%s" to "%s"', func.__name__, matchstr)
+        return func
+    return wrapper
```
I think I added this file back in by mistake, if so, kill it

Commit:
Removed Ears.py unneeded

Originally stripped because of a new layout of the files, but no longer
needed because of going back to original.

Commit:
New Code For Multiple Alias'

Commit:
update gitignore for plugins

Commit:
updated globals and special bot code. Plus a gitignore so plugins can be divided to dev

Commit:
Removed Answer Key, added alias options in settings

set global botname

Commit:
Dom's alias changes

Much better than mine

Commit:
Global Variables have been redone

Attributes are declared and stored in a dict

Commit:
test print

Commit:
path junk

Commit:
__init__

Commit:
Closer to original

Commit:
remove answer key and an update sh file

Commit:
small bash update

daredoes committed:
perms on sh

Commit:
its a bash script! HEADER

Commit:
better bash file

Commit:
gosh I'm a dumb dumb

Commit:
bash updates

binary1230 commented on slackbot/plugins/update.sh:
```diff
@@ -0,0 +1,18 @@
+      git pull
+      cd ..
+    done
+fi
```
I think we should do this in a layer outside of this repo, and keep this repo free of knowing anything about our specific implementation of updates/etc

daredoes replied:
I think knowing the function of updating is something that would be valuable to anyone setting up the bot. we can just put it in the git ignore

binary1230 commented on slackbot/plugins/upload.py:
```diff
@@ -1,16 +0,0 @@
-            download_file(url, tmpf)
-            message.channel.upload_file(fname, tmpf, 'downloaded from {}'.format(url))
-    elif url.startswith('/'):
-        message.channel.upload_file(fname, url)
```
re-add this file

daredoes replied:
it doesn't belong here

daredoes replied:
but fine.

binary1230 commented on slackbot/plugins/hello.py:
```diff
@@ -1,35 +0,0 @@
-
-@respond_to(u'你好')
-def hello_unicode_message(message):
-    message.reply(u'你好!')
```
restore this file

daredoes replied:
restored

binary1230 commented on .gitignore:
```diff
@@ -1,12 +1,6 @@
+
+.idea/
+
+magbot/
```
restore the existing stuff

binary1230 commented on run.py:
```diff
@@ -3,7 +3,6 @@
 import logging
 import logging.config
-from slackbot import settings
 from slackbot.bot import Bot
```
prob legit to keep this killed if nothing in the original implementation is using it

binary1230 commented on slackbot/bot.py:
```diff
@@ -7,22 +7,27 @@
 
 class Bot(object):
-    def __init__(self):
+    def __init__(self, settings, root):
```
any reason remaining for still passing in settings here?

binary1230 replied:
what is root?

daredoes replied:
by passing settings, you can have bot.py in the brain instead of in each bot folder.
Root is the folder the python script is launched from, used for updating commands.

binary1230 commented on slackbot/dispatcher.py:
```diff
@@ -100,18 +107,17 @@ def loop(self):
-        m = Message(self._client, msg)
-        m.reply(default_reply)
+            default_reply = '\n'.join(to_utf8(default_reply))
+        self._client.rtm_send_message(msg['channel'], default_reply)
```
not clear what the changes to this section are actually doing, can you elaborate?

daredoes replied:
Manually added in a help to the respond_to and listen_to commands, passed it by modifying p value from a single value to a (pair, value). For self.AT_MESSAGE_MATCHER its for bot specific matcher text. Pass the aliases through the specific bot settings that are fetched from plugins

binary1230 replied:
One thing, I think it's cleaner to keep the regex stuff where it is, we won't be running multiple bots out of one process so therthey're only needs to be one copy of it, and not a per-bot copy

Commit:
Moved bot launch from child folder to parent

Commit:
Merge remote-tracking branch 'origin/master' into develop

# Conflicts:
#	slackbot/dispatcher.py
The settings are passed through the plugin manager in the init. Removed
some useless imports. Message matcher inside the object so that the
alias’ are loaded from the settings passed down through the bot start,
not through an import statement. (Can move the settings file to other
locations for different implementations of updating the bot)

Commit:
does this work?

Commit:
Remove Useless imports and variables

Commit:
Doing little changes to make things ready for the pull request

migetman9 commented:
@binary1230 HEY. I did the thing where it can be merged with the master. Yaaaaaay!

Commit:
Moved the plugins folder as asked, removed root folder need for bot, modified how global attributes can be declared

Commit:
forgot this small commit about the root folder

Commit:
Alias Testing

Commit:
convert help back to original method, add utf-8 encoding for SRE patterns, add a use-case of the doc to hello plugin

Commit:
remove one parameter that is not needed

Commit:
restore the settings file with commented out variables, restore the plugins folder to its original state, restore the run file to its original state

binary1230 commented on run.py:
```diff
@@ -19,5 +19,6 @@ def main():
     bot = Bot()
     bot.run()
 
+
```
kill

binary1230 commented on slackbot/bot.py:
```diff
@@ -39,7 +45,6 @@ def _keepactive(self):
             self._client.ping()
 
-
 def respond_to(matchstr, flags=0):
```
kill whitespace

binary1230 commented on slackbot/bot.py:
```diff
@@ -48,7 +53,7 @@ def wrapper(func):
 
 
-def listen_to(matchstr, flags=0):
+def listen_to(matchstr,flags=0):
```
add whitespace back

binary1230 commented on slackbot/settings.py:
```diff
@@ -2,13 +2,17 @@
 # API_TOKEN = '###token###'
+# ALIASES = ",".join([r'!', r'$'])
+
+#default_reply = "Sorry, what was that?"
```
kill

binary1230 commented on plugins/.gitignore:
```diff
@@ -0,0 +1,4 @@
+[^.]*
+!__init__.py
+!upload.py
+!hello.py
```
need to restore `VERSION` file or else 'pip install' won't work anymore for slackbot when it's packaged up (or installed from docker)

I think we should scrap the idea of moving the plugins dir to the root dir in the core repo and just leave it where it is

binary1230 commented on slackbot/dispatcher.py:
```diff
@@ -6,18 +6,12 @@
-regex_aliases = '|'.join([re.escape(s) for s in settings.ALIASES.split(',')]) if hasattr(settings, 'ALIASES') else never_matches_anything
-
-AT_MESSAGE_MATCHER = re.compile(r'^(?:\<@(\w+)\>|({})):? (.*)$'.format(regex_aliases))
 
```
gotta merge in the new changes from the aliases_support branch

Commit:
re-add the github support for updating

I believe its a very good way of keeping the bot available for open
development. So far I’ve been able to spin up 3 bots without having to
modify and code. Having settings.py ignored, but leaving a
settings-example.py keeps the real settings entered by the user from
being added in by a future pull request. The method I’m using to update
kinda forcibly suggests using supervisord to manage the program. I’m
not quite sure how to get around this for now.

Commit:
Merge remote-tracking branch 'origin/develop' into develop

Moved the example settings into the slackbot folder. Changed the bot.py
back to the original code. No longer needed for multiple bots. Plugins
re-added for easy future PR’s

Commit:
change run.py back to original code

Commit:
remove reload.sh and add an example. reload.sh needs to be process specific

daredoes committed:
Update readme.md for people attempting to develop

This is supposed to help people. It may not properly communicate that. I accept all criticism.

Commit:
update install requirements

Commit:
get package ready for docker

daredoes committed:
Merge pull request #3 from magfest/enviroment-variables

get package ready for docker

Commit:
delete included package

url lib comes with python

Commit:
fix installs

Commit:
fix more installs

im so close

Commit:
add catches for new env variables

Commit:
super ready

binary1230 commented:
Just doing some branch cleanup 10years later? :)
